### PR TITLE
W-14488506: Provide an API for `mule-framework` diet tooling/operation executions

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/event/CoreEvent.java [hunk 1]
```diff
@@ -39,6 +42,21 @@
...
+   */
+  static CoreEvent nullEvent() {
+    return getNullEvent();
+  }
+
+  /**
+   * @param muleContext context on which the event will be associated.
+   * @return a dummy event associated on the given {@code muleContext} and no values.
+   */
+  static CoreEvent nullEvent(MuleContext muleContext) {
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/event/CoreEvent.java:56]: is this one relying on the muleContext really needed in the api?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/api/event/CoreEvent.java (hunk 1 above)
Comment (rbourbonnavarro) [reply core/src/main/java/org/mule/runtime/core/api/event/CoreEvent.java:56]: It's actually the use case we have in the `mule-framework`.

**Review by elrodro83 — APPROVED** [review]

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Pull request merged**